### PR TITLE
Use token from luigi

### PR DESCRIPTION
**Pull request by kwiatekus**
**Description**

Changes proposed in this pull request:

- clean up /app/auth folder - use only tokenInterceptor
- remove angular-oidc module from dependencies
- bootstrap angular app ONLY after init context received from luigi
- replace oidc.getToken() fn with reading token from Luigi (via luigi client)

### CODE REVIEW

File: core/src/app/navigation/header/header.component.ts [hunk 1]
```diff
@@ -111,16 +104,10 @@ export class HeaderComponent implements OnInit, OnDestroy {
   }
 
   get name() {
-    const claims = this.oauthService.getIdentityClaims();
-    if (!claims) {
-      return null;
-    }
-    return claims['name'];
+    return null;
```
Comment (parostatkiem) [inline core/src/app/navigation/header/header.component.ts]: What does it do?

File: core/src/app/navigation/header/header.component.ts [hunk 2]
```diff
@@ -111,16 +104,10 @@ export class HeaderComponent implements OnInit, OnDestroy {
...
   }
 
-  public logout() {
-    this.loginService.logout();
-  }
+  public logout() {}
```
Comment (parostatkiem) [inline core/src/app/navigation/header/header.component.ts]: What is it for?

File: core/src/app/shared/components/copy2clipboard-modal/copy2clipboard-modal.component.ts [hunk 3]
```diff
@@ -15,7 +14,7 @@ export class Copy2ClipboardModalComponent {
   private isCopied: false;
   private message: string;
 
-  constructor(private oAuthService: OAuthService) {}
+  constructor() {}
```
Comment (parostatkiem) [inline core/src/app/shared/components/copy2clipboard-modal/copy2clipboard-modal.component.ts]: Is it necessary?

File: core/src/app/auth/token.interceptor.spec.ts [hunk 4]
```diff
@@ -1,25 +1,28 @@
+import * as LuigiClient from '@kyma-project/luigi-client';
```
Comment (parostatkiem) [inline core/src/app/auth/token.interceptor.spec.ts]: why not `import LuigiClient from ...`? The same with other places.

File: core/src/app/navigation/header/header.component.ts (hunk 1 above)
Comment (parostatkiem) [inline core/src/app/navigation/header/header.component.ts]: Is it necessary? the same with below `logout()`...

File: core/src/app/shared/components/copy2clipboard-modal/copy2clipboard-modal.component.ts (hunk 3 above)
Comment (parostatkiem) [inline core/src/app/shared/components/copy2clipboard-modal/copy2clipboard-modal.component.ts]: I think you can remove it.

File: core/src/app/navigation/header/header.component.ts (hunk 1 above)
Comment (kwiatekus) [reply core/src/app/navigation/header/header.component.ts]: It was supposed to get user name and display it on the nav header - please note that header component will be removed in the cleanup task (already WIP, see : https://github.com/kwiatekus/console/commit/c8e319085b320f65c7d64f3d0e02312c19823cb3)

File: core/src/app/navigation/header/header.component.ts (hunk 2 above)
Comment (kwiatekus) [reply core/src/app/navigation/header/header.component.ts]: I simply wanted to remove the reference to loginService. The whole header component will be removed in the cleanup task (already WIP, see : kwiatekus@c8e3190)

File: core/src/app/auth/token.interceptor.spec.ts (hunk 4 above)
Comment (kwiatekus) [reply core/src/app/auth/token.interceptor.spec.ts]: when importing like this (either in interceptor or its test, or in other places): 
```
import LuigiClient from '@kyma-project/luigi-client';
```
the unit tests fail.

```
LOG: TypeError: Cannot read property 'getEventData' of undefined
TypeError: Cannot read property 'getEventData' of undefined
    at TokenInterceptor.intercept (webpack:///./src/app/auth/token.interceptor.ts?:23:48)
```

There might be an issue with luigi client bundling, but I wouldnt fix it within this task

File: core/src/app/shared/components/copy2clipboard-modal/copy2clipboard-modal.component.ts (hunk 3 above)
Comment (kwiatekus) [reply core/src/app/shared/components/copy2clipboard-modal/copy2clipboard-modal.component.ts]: done

**Pull request merged**